hpe-dev-incubator/hpe-dev-portal
# Blogs page - The Load More option does not bring you back to where you were after reading a blog

## Issue
Issue: https://github.com/hpe-dev-incubator/hpe-dev-portal/issues/219
When a user selects a Blog Card which will take them to the Blog Post page, then goes back to the Blog page, they will be automatically taken to the top of the Blog page. 

## Description - A brief explanation of what this pull request implements.
- When user is on a Blog Post page, clicking the browser back button will take the user back to the same spot on the Blog page.
- When user is on a Blog Post page, scrolling all the way down and clicking on **Go to Blog Page**, will take the user to the same spot on the Blog page
- If user clicks on **Blog** from the Header Navigation, user will be taken to the top of the Blog page

## Proposed Changes - What does this PR add/edit/remove to achieve its purpose
- Uses localStorage to store **latestPage** and **latestBlogPosts** data so that blogs that have been generated from the **Load More** button will already be rendered. 
- When a Blog Card is clicked, the window Y position will be saved to localStorage. When the Blog page is rendered, it will look for the window y position from localStorage and then scroll to that position. There is a set timeout because blog data is loading from localStorage. 
- Created a `gatsby-browser.js` to save previous paths. This is used for when a user clicks on the **Blog** from the Header Navigation. If the previous path does not include `/blog`, then it will automatically scroll to the top. I'm assuming that when a user clicks on this, they want to be automatically taken to the top, but I'm happy to talk about this more. 


## Tests

### Test back button browser
1. Go to Blog page
2. Click on Blog card
3. When you are on the Blog Post page, click the browser back button to go back
4. You should be taken to the same spot that you were perviously on in the Blog page

### Test **Go to Blog Page**
1. Go to Blog page
2. Click on Blog card
3. When you are on the Blog Post page, scroll down to the bottom and click on **Go to Blog Page**
4. You should be taken to the same spot that you were perviously on in the Blog page

### Test **Blog** on the Header
1. On any page other than Blog page, click on the **Blog** button on the Header Navigation
2. You should be taken to the top of the Blog page
